app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java

```diff
@@ -110,25 +108,43 @@ public void start(Stage primaryStage) throws IOException {
       return;
     }
 
-    Stopwatch fxmlLoadTimer = Stopwatch.createStarted();
-
-    FXMLLoader loader = new FXMLLoader(MainWindowController.class.getResource("MainWindow.fxml"));
-    mainPane = loader.load();
-    long fxmlLoadTime = fxmlLoadTimer.elapsed(TimeUnit.MILLISECONDS);
-    logger.log(fxmlLoadTime >= 500 ? Level.WARNING : Level.INFO, "Took " + fxmlLoadTime + "ms to load the main FXML");
-    final MainWindowController mainWindowController = loader.getController();
-
-    primaryStage.setScene(new Scene(mainPane));
-
+    notifyPreloader(new ShuffleboardPreloader.StateNotification("Loading base plugin", 0.125));
     PluginLoader.getDefault().load(new BasePlugin());
 
     Recorder.getInstance().start();
 
+    notifyPreloader(new ShuffleboardPreloader.StateNotification("Loading NetworkTables plugin", 0.25));
     PluginLoader.getDefault().load(new NetworkTablesPlugin());
+    notifyPreloader(new ShuffleboardPreloader.StateNotification("Loading CameraServer plugin", 0.375));
     PluginLoader.getDefault().load(new CameraServerPlugin());
+    notifyPreloader(new ShuffleboardPreloader.StateNotification("Loading Powerup plugin", 0.5));
     PluginLoader.getDefault().load(new PowerupPlugin());
+    notifyPreloader(new ShuffleboardPreloader.StateNotification("Loading custom plugins", 0.625));
     PluginLoader.getDefault().loadAllJarsFromDir(Storage.getPluginPath());
+    notifyPreloader(new ShuffleboardPreloader.StateNotification("Loading custom plugins", 0.75));
     PluginCache.getDefault().loadCache(PluginLoader.getDefault());
+    notifyPreloader(new ShuffleboardPreloader.StateNotification("Loading custom plugins", 0.875));
+    Thread.sleep(250); // wait to let the new status be visible before overwriting it
+    notifyPreloader(new ShuffleboardPreloader.StateNotification("Starting up", 1));
+    Thread.sleep(20); // small wait to let the status be visible - the preloader doesn't get notifications for a bit
```

Comment:
Every second counts.

Reply:
Without a delay here, the final status will never be shown. It'll be stuck showing "Loading custom plugins" when that's already been completed

Reply:
Ok